docs/upgrading_to_v4.0.md

```diff
@@ -0,0 +1,43 @@
+# Upgrading to v4.0
+The v4.0 release of
+*terraform-google-modules/gcloud/google//modules/kubectl-wrapper*
+is a backwards incompatible release.
+
+### For `use_existing_context = true` use `terraform-google-modules/gcloud/google`:
+
+```diff
+  module "kubectl_existing_context" {
+-   source = "terraform-google-modules/gcloud/google//modules/kubectl-wrapper"
+-   version               = "~> 3.0"
+-   use_existing_context  = true
++   source                = "terraform-google-modules/gcloud/google"
```

Comment:
Using the gcloud root module to load a kubectl context looks a bit out of place. I'm wondering we should keep this the way it is and may be add another module for fleets.

Reply:
Thanks @g-awmalik - Yeah, there are some tradeoffs.  Currently I suspect that `use_existing_context` only works if values for the (optional) `cluster_name`, `cluster_location`, and `project_id` are provided, as otherwise the script won't receive the correct number of aguments.

An alternative approach I considered would be to actuate `create_cmd_entrypoint` as `use_existing_context` doesn't actually need to use the `kubectl_wrapper.sh` script.  However my preference would be to specifically require the minimum necessary variables, which is complicated when they aren't required for `use_existing_context`.  A related permutation might be to make a new `kubectl` submodule, which is specially design for existing content (no wrapper, no name, location, etc).  This might be the "ideal" option, but would still be a breaking change/migration required.

I also thought about creating a "kubectl-fleet-wrapper" (or similar), which would be very similar (although not identical) to the current implementation.  That said, it would be the best for maximum backward compatibility.

We can discuss further directly.

Reply:
>> I also thought about creating a "kubectl-fleet-wrapper" (or similar), which would be very similar (although not identical) to the current implementation. That said, it would be the best for maximum backward compatibility.

I was leaning towards this as well. We can discuss further 1-1.

Reply:
I might be missing some context but what is the problem with keeping `use_existing_context`?

Reply:
Thanks Everyone! Based on the discussion we'll go with a fleet specific sub-module to preserve compatibility.